### PR TITLE
Set dialog title to OS release

### LINKED ISSUE #64: Set title to "Installing OS_NAME"
...with `OS_NAME` being the name of the OS from LSB.

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/48739283-set-title-to-installing-os_name?utm_campaign=plugin&utm_content=tracker%2F46189108&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F46189108&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by danrabbit**
fixes #64

### CODE REVIEW

File: src/MainWindow.vala [hunk 1]
```diff
@@ -69,4 +70,29 @@ public class Installer.MainWindow : Gtk.Dialog {
             stack.set_visible_child_name ("progress-view");
         });
     }
+
+    private string get_os_release () {
```
Comment (donadigo) [inline src/MainWindow.vala]: This method should be static as it doesn't access anything from the class instance.

File: src/MainWindow.vala [hunk 2]
```diff
@@ -69,4 +70,29 @@ public class Installer.MainWindow : Gtk.Dialog {
...
+        string os;
+        var file = File.new_for_path("/etc/os-release");
```
Comment (donadigo) [inline src/MainWindow.vala]: Can this `/etc/os-release` be a constant?

Also, spaces between braces `new_for_path ()`.

File: src/MainWindow.vala [hunk 3]
```diff
@@ -69,4 +70,29 @@ public class Installer.MainWindow : Gtk.Dialog {
...
+
+        try {
+            var osrel = new Gee.HashMap<string, string> ();
```
Comment (donadigo) [inline src/MainWindow.vala]: If this is the only information we're gonna get from that file, this `HashMap` isn't needed here and we can just assign the OS string in the `while` statement and then `break` after we're done. If that's not the case we should probably make a new class to parse and provide the needed information about the OS to not clutter the `MainWindow` class.

File: src/MainWindow.vala (hunk 2 above)
Comment (Conan-Kudo) [reply src/MainWindow.vala]: This is basically constant. This path never changes.

File: src/MainWindow.vala [hunk 4]
```diff
@@ -69,4 +70,29 @@ public class Installer.MainWindow : Gtk.Dialog {
...
+                var osrel_component = line.split ("=", 2);
+                if ( osrel_component.length == 2 ) {
+                    osrel[osrel_component[0]] = osrel_component[1].replace ("\"", "");
+                }
+            }
+
+            os = osrel["PRETTY_NAME"];
+        } catch (Error e) {
+            warning("Couldn't read os-release file, assuming elementary OS");
+            os = "elementary OS";
```
Comment (Conan-Kudo) [inline src/MainWindow.vala]: I would suggest assuming `Generic OS` or something unbranded.

File: src/MainWindow.vala (hunk 2 above)
Comment (Conan-Kudo) [reply src/MainWindow.vala]: This is basically constant. This path never changes.

Comment (donadigo) [reply src/MainWindow.vala]: I mean, to have this declared in an a `const string`.

Comment (Conan-Kudo) [reply src/MainWindow.vala]: It should be a `const string` because it's basically constant, I mean. :)

**Pull request merged**